lib/charms/opensearch/v0/opensearch_relation_peer_cluster.py

```diff
@@ -806,11 +872,11 @@ def _cm_nodes(self, orchestrators: PeerClusterOrchestrators) -> List[Node]:
             if rel_id == -1:
                 continue
 
-            data = self.get_obj_from_rel(key="data", rel_id=rel_id)
-            if not data:  # not ready yet
+            redacted_dict_str = self.get_from_rel(key="data", rel_id=rel_id, remote_app=True)
+            if not redacted_dict_str:  # not ready yet
                 continue
 
-            data = PeerClusterRelData.from_dict(data)
+            data = rel_data_from_redacted_str(self.charm, redacted_dict_str)
```

Comment:
You could use the PeerClusterRelData.from_dict() method here to create the structure. Therefore return the updated data from `rel_data_from_redacted_str()` to here and load it via the PeerClusterRelData.from_dict() method.